scripts/irods/test/test_access_time_updates.py

```diff
@@ -0,0 +1,249 @@
+import json
+import time
+import unittest
+
+from pathlib import Path
+
+from . import session
+from .. import lib
+from .. import paths
+from .. import test
+from ..controller import IrodsController
+from ..exceptions import IrodsError
+
+rodsadmins = [('otherrods', 'rods')]
+rodsusers  = [('alice', 'apass')]
+
+
+class Test_Access_Time_Updates(session.make_sessions_mixin(rodsadmins, rodsusers), unittest.TestCase):
+
+    def setUp(self):
+        super(Test_Access_Time_Updates, self).setUp()
+
+        self.admin = self.admin_sessions[0]
+        self.user = self.user_sessions[0]
+
+    def tearDown(self):
+        super(Test_Access_Time_Updates, self).tearDown()
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY,
+        "Topology testing for a GenQuery column isn't necessary because GenQuery always redirects to the Catalog Service Provider")
+    def test_GenQuery_supports_access_time_issue_8260(self):
+        data_object = 'test_GenQuery_supports_access_time_issue_8260.txt'
+        self.user.assert_icommand(['itouch', data_object])
+
+        query = f"select DATA_ACCESS_TIME where DATA_NAME = '{data_object}'"
+
+        with self.subTest('using GenQuery1'):
+            self.user.assert_icommand(['iquest', query], 'STDOUT', r'DATA_ACCESS_TIME = 0\d{10}', use_regex=True)
+
+        with self.subTest('using GenQuery2'):
+            self.user.assert_icommand(['iquery', query], 'STDOUT', r'"0\d{10}"', use_regex=True)
+
+    def test_access_time_matches_modification_time_for_new_replicas__issue_8260(self):
+        def assert_atime_matches_mtime_for_replica(data_object, replica_number):
+            query = f"select DATA_ACCESS_TIME, DATA_MODIFY_TIME where DATA_NAME = '{data_object}' and DATA_REPL_NUM = '{replica_number}'"
```

Comment:
Had to change this test because the icommands do not provide enough precision for us to assert atime and mtime match exactly.

Having the atime and mtime match wasn't a requirement or goal of the atime implementation either. The goal of this test was to show that the atime reflects a recent timestamp for newly created replicas and data objects. I've tweaked the test to reflect that goal better.

Will push the changes up as soon as the topology tests for consumer completes. There's one test failure that I'll need to work out first though.

Reply:
Done.